### PR TITLE
fix/ADF-1827/add-sync-by-id

### PR DESCRIPTION
**Pull request by shaveko**
related to https://oat-sa.atlassian.net/browse/ADF-1827

### Description

Adds syncById method to api of TranslationSyncService

### How to test 

checkout related taoTests branches:
 - https://github.com/oat-sa/extension-tao-test/tree/fix/ADF-1827/sync-translations-on-authoring
 - https://github.com/oat-sa/extension-tao-testqti/tree/fix/ADF-1827/remove-translation-sync-call

#### Steps to reproduce / check if issue is fixed:
 
 - create an item
 - create test 
 - add item to the test 
 - create test translation to some language, save it
 - add translation to the item in that language
 - go back to tests and edit the translation of the test you have created

label should exist for the item

### CODE REVIEW

File: models/classes/Translation/Service/TranslationSyncService.php [hunk 1]
```diff
@@ -63,6 +63,11 @@ public function syncByRequest(ServerRequestInterface $request): core_kernel_clas
         $requestParams = $request->getParsedBody();
         $id = $requestParams['id'] ?? null;
 
+        return $this->syncById($id);
+    }
+
+    public function syncById(string $id): core_kernel_classes_Resource
```
Comment (gabrielfs7) [inline models/classes/Translation/Service/TranslationSyncService.php:69]: @shaveko the FE already calls and endpoint `translation/sync` to call the `syncByRequest`. I am not sure also asking the BE to do the same is a good idea 🤔

Comment (jsconan) [reply models/classes/Translation/Service/TranslationSyncService.php:69]: @gabrielfs7 actually, the FE call is removed, see: https://github.com/oat-sa/extension-tao-testqti/pull/2539/files

I don't know the rational though

Comment (gabrielfs7) [reply models/classes/Translation/Service/TranslationSyncService.php:69]: I missed the PR with FE changes removed, I thought we were keeping both, which would be expensive. Now all good. Thanks

**Review by gabrielfs7 — APPROVED** [review]
LGTM, Good job!

Code review only

- [] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by jsconan — APPROVED** [review]
Code looks good, and it runs fine.

One note, though, when I add an item to the main test, it does not show up in the translated test.

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code respects code style rules
- [x] New code respects best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

**Pull request merged**